src/test/kotlin/com/terraformation/backend/tracking/ObservationServiceTest.kt

```diff
@@ -619,8 +621,35 @@ class ObservationServiceTest : DatabaseTest(), RunsAsUser {
     }
 
     @Test
-    fun `throws exception rescheduling an observation if the observation is not currently in Overdue state`() {
-      val observationId = insertObservation(state = ObservationState.Upcoming)
+    fun `throws exception rescheduling an observation if the observation is Completed`() {
+      val observationId =
+          insertObservation(state = ObservationState.Completed, completedTime = Instant.EPOCH)
+
+      val startDate = LocalDate.EPOCH
+      val endDate = startDate.plusDays(1)
+
+      assertThrows<ObservationRescheduleStateException> {
+        service.rescheduleObservation(observationId, startDate, endDate)
+      }
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+        strings =
+            [
+                "InProgress",
+                "Overdue",
+            ])
+    fun `throws exception rescheduling an In-Progress or Overdue observation if there are observed plots`(
+        stateName: String
+    ) {
+      insertPlantingSite()
+      insertPlantingZone(numPermanentClusters = 1, numTemporaryPlots = 1)
+      insertPlantingSubzone()
+      insertMonitoringPlot()
+
+      val observationId = insertObservation(state = ObservationState.valueOf(stateName))
+      insertObservationPlot(completedTime = Instant.EPOCH)
```

Comment:
This will probably fail; you need to set both the claimed and completed fields. If you do `insertObservationPlot(claimedBy = user.userId, completedBy = user.userId)` it will fill in the timestamps for you.

That said, maybe it makes sense to automatically set claimed by/time here if completed by/time are set; the way it is now is extra verbosity for no real benefit.

Reply:
Sounds good, will make that change. This test passed btw, but the change does make sense.. will update.